push_messages/db.py

```diff
@@ -39,21 +34,23 @@ def create_key_table(tablename):
     return table
 
 
-def make_table_or_return(tablename):
+def make_table_or_return(dynamodb, tablename):
     table_names = [x.name for x in dynamodb.tables.all()]
     if tablename in table_names:
         return dynamodb.Table(tablename)
     else:
-        return create_key_table(tablename)
+        return create_key_table(dynamodb, tablename)
 
 
 class KeyResource(object):
-    def __init__(self, tablename, autocreate=True):
+    def __init__(self, tablename, autocreate=True, db_options=None):
+        db_opts = db_options or {}
+        self._dynamodb = boto3.resource("dynamodb", **db_opts)
```

Comment:
If you are passing this as an argument, do you need to slot it as well? I'm not seeing references to it outside of this function.